ocaml/xapi-idl/cluster/cluster_interface.ml

```diff
@@ -15,12 +15,45 @@ type debug_info = string [@@deriving rpcty]
 (** Name of the cluster *)
 type cluster_name = string [@@deriving rpcty]
 
-(** An IPv4 address (a.b.c.d) *)
-type address = IPv4 of string [@@deriving rpcty]
+type ipv4v6 = IPv4 of string | IPv6 of string [@@deriving rpcty]
 
-let printaddr () = function IPv4 s -> Printf.sprintf "IPv4(%s)" s
+let string_of_ip = function IPv4 a | IPv6 a -> a
 
-let str_of_address address = match address with IPv4 a -> a
+type uuid_addr = {ip: ipv4v6; hostid: string; hostname: string}
+[@@deriving rpcty]
+
+type address = IPv4 of string | UuidAddr of uuid_addr [@@deriving rpcty]
```

Comment:
Is IPv6 not allowed here?

Reply:
It is allowed inside the uuid_addr. In the future, we could potentially drop the `IPv4` variant altogether.

Reply:
Sorry, I didn't get the point before.

Reply:
This indeed looks strange: would expect IPv4, IPv6, and UUID when we use IPv6 in other places.

Reply:
I agree. I think the problem is backward compatibility again, I think. If we had three variants, then we can't transfer uuid along with ip address, and both are necessary for this patch to work.

Reply:
I think the name `UuidAddr` is not very clear, as you wouldn't think that it has IP and hostname info in it as well. Perhaps `Extended` or something like that?